tests/rados/test_osd_inconsistency_pg.py

```diff
@@ -0,0 +1,110 @@
+
+
+def run(ceph_cluster, **kw):
+    log.info(run.__doc__)
+    config = kw["config"]
+    cephadm = CephAdmin(cluster=ceph_cluster, **config)
+    rados_obj = RadosOrchestrator(node=cephadm)
+    client_node = ceph_cluster.get_nodes(role="client")[0]
+    pool_obj = PoolFunctions(node=cephadm)
+    try:
+        log.info("Creating the pool replicated pool")
+        create_pools(config, rados_obj, client_node)
+        log.info("Pool creation completed")
+        pool_name = config["create_pools"][0]["create_pool"]["pool_name"]
+
+        # Adding an object
+        oname = "UNIQUEOBJECT{i}".format(i=random.randint(0, 10000))
+        log.info(f"Creating the object-{oname} in the pool-{pool_name}")
+        pool_obj.do_rados_put(client=client_node, pool=pool_name, obj_name=oname)
+        log.info(f"The object-{oname} is created in the pool-{pool_name}")
+        osd_map_output = rados_obj.get_osd_map(pool=pool_name, obj=oname)
+        primary_osd = osd_map_output["acting_primary"]
+        log.info(f"The object stored in the primary osd number-{primary_osd}")
+        pg_id = osd_map_output["pgid"]
+        log.info(f"The object {oname} is created in the pg-{pg_id}")
+        # Adding the key value to object
+        log.info("Adding a key to the object")
+        cmd_setkey = f"rados -p {pool_name} setomapval {oname} key0 value0"
+        client_node.exec_command(cmd=cmd_setkey, sudo=True)
+        # stoping the OSD
+        if not rados_obj.change_osd_state(action="stop", target=primary_osd):
+            log.error(f"Unable to stop the OSD : {primary_osd}")
+            raise Exception("Execution error")
+        acting_osd_node = rados_obj.fetch_host_node(
+            daemon_type="osd", daemon_id=primary_osd
+        )
+        cmd_base = f"cephadm shell --name osd.{primary_osd} --"
+        log.info(f"Corrupting the object-{oname} key.")
+        cmd_get_snaObj = (
+            f"{cmd_base} ceph-objectstore-tool --data-path /var/lib/ceph/osd/ceph-"
+            f"{primary_osd} --pgid {pg_id} {oname} set-omap key_corrupt1 /etc/hosts"
+        )
+        acting_osd_node.exec_command(sudo=True, cmd=cmd_get_snaObj)
+        if not rados_obj.change_osd_state(action="start", target=primary_osd):
+            log.error(f"Unable to start the OSD : {primary_osd}")
+            raise Exception("Execution error")
+            # Wait until pg active+clean state
+        while True:
+            pg_report = rados_obj.check_pg_state(pgid=pg_id)
+            if "active+clean" in pg_report:
+                break
+            log.info(f"The pg-{pg_id} is not in active+clean state")
+            time.sleep(10)
+        log.info(f"The pg-{pg_id} is in active+clean state")
+        # Perform deep-scrub
+        log.info(f"Performing the deep-scrub on the pg-{pg_id}")
+        rados_obj.run_deep_scrub(pgid=pg_id)
+        # sleeping for 10 seconds, waiting for scrubbing to be over
+        time.sleep(10)
+        # Checking the cluster health status
```

Comment:
Please write a method for running health status and list-inconsistent-pg on a pool so that we can reuse the same in other tests for inconsistency.

Reply:
Method added.